pkg/server/server.go

```diff
@@ -399,10 +395,15 @@ func (s *ProxyServer) Proxy(stream client.ProxyService_ProxyServer) error {
 				return
 			}
 
-			if s.warnOnChannelLimit && len(recvCh) >= xfrChannelSize {
+			select {
+			case recvCh <- in: // Send didn't block, carry on.
+			default: // Send blocked; record it and try again.
 				klog.V(2).InfoS("Receive channel on Proxy is full", "userAgent", userAgent, "serverID", s.serverID)
```

Comment:
I think "on Proxy" here is a bit obscure. Might be more clear if we referred to is as "forwarding traffic from the frontend"?